plugins/modules/omevv/omevv_firmware.py

```diff
@@ -0,0 +1,795 @@
+            device information.
+        type: str
+requirements:
+  - "python >= 3.9.6"
+author:
+  - "Rajshekar P(@rajshekarp87)"
+attributes:
+    check_mode:
+        description: Can run in check_mode and return changed status prediction without modifying target, if not supported the action will be skipped.
+        support: full
+    diff_mode:
+        description: Will return details on what has changed (or possibly needs changing in check_mode), when in diff mode.
+        support: full
+notes:
+    - Run this module from a system that has direct access to Dell OpenManage Enterprise.
+    - This module supports IPv4 and IPv6 addresses.
+"""
+
+EXAMPLES = r"""
+---
+- name: Immediately update the firmware of a single component for a specific host
+  dellemc.openmanage.omevv.omevv_firmware:
+    hostname: "192.168.0.1"
+    vcenter_uuid: "xxxxx"
+    vcenter_username: "username"
+    vcenter_password: "password"
+    ca_path: "path/to/ca_file"
+    run_now: false
+    date_time: "2024-09-10T20:50:00Z"
+    enter_maintenance_mode_timeout: 60
+    enter_maintenance_mode_options: FULL_DATA_MIGRATION
+    drs_check: true
+    evacuate_VMs: true
+    exit_maintenance_mode: true
+    reboot_options: NEXTREBOOT
+    maintenance_mode_count_check: true
+    check_vSAN_health: true
+    reset_idrac: true
+    delete_job_queue: true
+    targets:
+      - servicetag: SVCTAG1
+        firmware_components:
+          - "DCIM:INSTALLED#802__Diagnostics.Embedded.1:LC.Embedded.1"
+
+- name: Update the firmware of multiple components at scheduled time for a specific host
+  dellemc.openmanage.omevv.omevv_firmware:
+    hostname: "192.168.0.1"
+    vcenter_uuid: "xxxxx"
+    vcenter_username: "username"
+    vcenter_password: "password"
+    ca_path: "path/to/ca_file"
+    run_now: false
+    date_time: "2024-09-10T20:50:00+05:30"
+    enter_maintenance_mode_timeout: 60
+    enter_maintenance_mode_options: ENSURE_ACCESSIBILITY
+    drs_check: true
+    evacuate_VMs: true
+    exit_maintenance_mode: true
+    reboot_options: FORCEREBOOT
+    maintenance_mode_count_check: true
+    check_vSAN_health: true
+    reset_idrac: false
+    delete_job_queue: false
+    targets:
+      - host: 192.168.0.2
+        firmware_components:
+          - "DCIM:INSTALLED#iDRAC.Embedded.1-1#IDRACinfo"
+          - "DCIM:INSTALLED#301_C_BOSS.SL.14-1"
+          - "DCIM:INSTALLED#807__TPM.Integrated.1-1"
+
+- name: Retrieve firmware compliance report of all the hosts in the specific cluster
+  dellemc.openmanage.omevv_firmware_compliance_info:
+    hostname: "192.168.0.1"
+    vcenter_uuid: "xxxxx"
+    vcenter_username: "username"
+    vcenter_password: "password"
+    ca_path: "path/to/ca_file"
+    clusters:
+      - cluster_name: cluster_a
+  register: compliance_data
+
+- name: Extract source name of all components
+  set_fact:
+    source_names: "{{ compliance_data.hostComplianceReports[0].componentCompliances|json_query('*.sourceName') }}"
```

Comment:
avoid using json_query in example as it's part of common.general collection, please update the example

Reply:
Please change this as it adds dependencies on community.general collection.

Reply:
Incorporated

Reply:
Incorporated